bam_qc_metrics/bam_qc.py

```diff
@@ -112,9 +112,18 @@ def __init__(self, config):
         self.slow_metrics = None
         (self.tmpdir, self.tmp_object) = self.setup_tmpdir(config[self.CONFIG_KEY_TEMP_DIR])
         self.unmapped_excluded_reads = None
+        # find metrics; if an error occurs, do logging and cleanup before exit
+        try:
+            self._find_metrics(config[self.CONFIG_KEY_BAM])
+        except:
+            self.logger.error("Unexpected error: %s", sys.exc_info()[0])
```

Comment:
Python logger has the `exception` function to log the trace:
https://stackoverflow.com/questions/5191830/how-do-i-log-a-python-error-with-debug-information
https://docs.python.org/3/library/logging.html#logging.Logger.exception

Reply:
Aha, useful. I'll do that.